report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessorTest.java

```
@@ -330,6 +330,14 @@ public void itShouldWritePdfContentsToTemporaryStorage() throws Exception {
         verify(temporaryStorage).writeChunk(eq(payload.getReportId()), eq(0), any(InputStream.class));
     }
 
+    @Test
+    public void itShouldHandleNullAssessmentTypeWhenGeneratingStudentChunk() {
+        when(icaReportRepository.findAllForStudentsByExamFilter(any(User.class), anyCollection(), argThat(p -> p.getAssessmentTypeCode() == null)))
+                .thenThrow(IllegalArgumentException.class);
+
+        writer.generateStudentChunkReport(message);
```

This works and the test passes with the fix and didn't before the fix, but I'm getting the following mockito error that I'm guessing will impact CI.

```
org.mockito.exceptions.misusing.UnnecessaryStubbingException: 
Unnecessary stubbings detected in test class: StudentReportProcessorTest
Clean & maintainable test code requires zero unnecessary code.
Following stubbings are unnecessary (click to navigate to relevant line of code):
  1. -> at org.opentestsystem.rdw.reporting.processor.stream.StudentReportProcessorTest.itShouldHandleNullAssessmentType(StudentReportProcessorTest.java:348)
Please remove unnecessary stubbings or use 'lenient' strictness. More info: javadoc for UnnecessaryStubbingException class.
```

In the setup we have a mock for this method already that accepts any ExamQueryParams and returns a HashMap 

In setup:
```
when(icaReportRepository.findAllForStudentsByExamFilter(any(User.class), anyCollection(), any(ExamQueryParams.class)))
                .thenReturn(new HashMap<>());
```

Based on this: https://stackoverflow.com/questions/2684630/how-can-i-make-a-method-return-an-argument-that-was-passed-to-it?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa I tried to make the return type conditional so there was only a single definition for this mock method but couldn't get it to work:

```
when(icaReportRepository.findAllForStudentsByExamFilter(any(User.class), anyCollection(), any(ExamQueryParams.class)))
                .thenReturn(new Answer<Map<Long, Map<Subject, IcaSummativeReport>>>() {
                    @Override
                    public Map<Long, Map<Subject, IcaSummativeReport>> answer(InvocationOnMock invocation) throws Throwable {
                        Object[] args = invocation.getArguments();

                        if (((ExamQueryParams)args[ 2 ]).getAssessmentTypeCode() == null)
                            throw new IllegalArgumentException();

                        return new HashMap<Long, Map<Subject, IcaSummativeReport>>();
                    }
                });
```

---

Hmm... rather than mocking out a failing case and assuring that it isn't hit, I'd probably confirm that it *is* doing what you want, adding something like:
```
        verify(icaReportRepository).findAllForStudentsByExamFilter(
                any(User.class),
                anyCollection(),
                examFilterCaptor.capture()
        );
        assertThat(examFilterCaptor.getValue().getAssessmentTypeCode()).isNotBlank();
```
to the existing repository interactions